### PR TITLE
This skips prepending a db name to the "table" when a query uses a subquery as the FROM

### PR DESCRIPTION
**Pull request by trobrock**
Simplest fix for https://app.circleci.com/pipelines/github/huntresslabs/portal/117577/workflows/a7988c6b-7cf1-44b5-a6ef-49411381f5b1/jobs/791202

### CODE REVIEW

File: spec/single/schema_creation_spec.rb [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    end
+
+    context 'when subquery does not contain FROM clause' do
+      it 'does not modify subquery without FROM' do
+        subquery = 'select 1 as id'
+        original_subquery = subquery.dup
+        schema_creation.send(:assign_database_to_subquery!, subquery)
+        expect(subquery).to eq(original_subquery)
+      end
+    end
```
Comment (kbennoune) [inline spec/single/schema_creation_spec.rb:89]: Do all these test add value? Seems like a lot are edge cases.

File: lib/active_record/connection_adapters/clickhouse/schema_creation.rb [hunk 2]
```diff
@@ -80,6 +80,9 @@ def assign_database_to_subquery!(subquery)
           # If you do not specify a database explicitly, ClickHouse will use the "default" database.
           return unless subquery
 
+          # Skip processing if FROM is followed by a subquery (indicated by opening parenthesis)
+          return if subquery.match(/from\s*\(/i)
```
Comment (kbennoune) [inline lib/active_record/connection_adapters/clickhouse/schema_creation.rb:84]: I do not like this at all, but I can't see a better way.

**Review by kbennoune — APPROVED** [review]
